core/src/main/java/org/mule/runtime/core/routing/AbstractSelectiveRouter.java

```diff
@@ -291,8 +291,7 @@ public void setRouterStatistics(RouterStatistics routerStatistics) {
 
   @Override
   public String toString() {
-    return format("%s [flow-construct=%s, started=%s]", getClass().getSimpleName(),
-                  flowConstruct != null ? flowConstruct.getName() : null, started);
+    return format("%s [flow=%s, started=%s]", getClass().getSimpleName(), getLocation().getParts().get(0).getPartPath(), started);
```

Comment:
Refactor repeated use of this type of thing into static helpers in ComponentLocation.